src/client/signatures/canonical_json.rs

```diff
@@ -1,5 +1,433 @@
-use serde_json;
+use serde_json::json;
+use regex::Regex;
+use regex::Captures;
+use log::{debug};
 
-pub fn serialize(input: &serde_json::value::Value) -> String {
-    format!("{}", input)
-}
\ No newline at end of file
+fn format_number(number: f64, lower_bound: f64, upper_bound: f64) -> String {
+    let regex = Regex::new(r"(\d+)e(\d+)").expect("error creating regex");
+
+    debug!("in format_number function");
+
+    if (0.0 < number && number < upper_bound) || (number >= lower_bound) {
+        debug!("converting number to scientific notation");
+        let number_string = format!("{:e}", number);
+        let string_num = regex.replace(&number_string, |caps: &Captures| {
+            format!("{}e+{}", &caps[1], &caps[2])
+        }).replace(r"\.?0+$", "");
+
+        return String::from(string_num);
+    }
+
+    debug!("returning number without scientifc notation");
+    return format!("{}", number).replace(r"\.?0+$", "");
+}
+
+fn serialize_child(child: &serde_json::Value) -> String {
+
+    let serialized_child_string = serialize(&child);
+
+    if child.is_string() {
+        return "\"".to_owned() + &serialized_child_string + "\"";
```

Comment:
What happens to quotes in the serialized child?  Are they already quoted?

Reply:
no, they are not quoted